### ISSUE TITLE
Trigger Order Issue

### ISSUE THREAD

**Issue by kasahh** [gateio/gateapi-python#119]
How do I create a "Trigger Market Take Profit Order" for a short trade? I tried posting these params for long and short:

For TP Long:
futures_price_triggered_order = FuturesPriceTriggeredOrder(initial = {"contract":symbol,"auto_size":"close_long","close":False,"is_close":False,"is_reduce_only":True,"reduce_only":True,"size":0,"price":str(0),"tif":"ioc"}, trigger ={"strategy_type":0,"price_type":1,"price":str(price),"rule":1,"expiration":0})

For TP Short:
futures_price_triggered_order = FuturesPriceTriggeredOrder(initial = {"contract":symbol,"auto_size":"close_short","close":False,"is_close":False,"is_reduce_only":True,"reduce_only":True,"size":0,"price":str(0),"tif":"ioc"}, trigger ={"strategy_type":0,"price_type":1,"price":str(price),"rule":2,"expiration":0})

But i end up getting two orders with same side of: "close long position". 

<img width="1345" alt="Screenshot 2022-06-06 at 16 41 42" src="https://user-images.githubusercontent.com/83571313/172257459-05d9f023-8229-454b-99b3-8d4f889ece4b.png">

I should be having "close long position" and "close short position" instead like on the website.

What I have tried so far:
- I initially left rule for TP Short at 1 but it gave an error about trigger price must be greater than price but this shouldn't be since i want to close a short trade whose take profit is less than current price. So i ended up using rule:2
- I included order_type = "close_long_position" and order_type = "close_short_position" for both  orders respectively but i got an error about order_type being an invalid argument or parameter or something like that.

What could I be doing wrong?

**Comment by revilwang** [comment]
`initial` and `trigger` fields are of `FuturesInitialOrder` and `FuturesPriceTrigger` types. You need to initialize two objects first then set them into the `FuturesPriceTriggeredOrder`

**Comment by kasahh** [comment]
I am unsure what you mean. I am still relatively new to python programming. Anyway, I tried to reproduce what I thought you meant by doing this:

trigger ={"strategy_type":0,"price_type":1,"price":str(price),"rule":1,"expiration":0}
initial = {"contract":symbol,"auto_size":"close_long","close":False,"is_close":False,"is_reduce_only":True,"reduce_only":True,"size":0,"price":str(0),"tif":"ioc"}
order_type = "close-long-order"
futures_price_triggered_order = FuturesPriceTriggeredOrder(initial,order_type, trigger)
data = api_response = api_instance.create_price_triggered_order(settle, futures_price_triggered_order)
print(data)


and i got this error as a response:

Reason: Bad Request
HTTP response headers: HTTPHeaderDict({'Date': 'Sun, 12 Jun 2022 10:34:56 GMT', 'Content-Type': 'application/json', 'Content-Length': '142', 'Connection': 'keep-alive', 'Server': 'openresty'})
HTTP response body: {"label":"AUTO_INVALID_REQUEST_BODY","message":"invalid json json: cannot unmarshal object into Go struct field TouchOrder.id of type uint64"}


Is it possible to give a more precise way of successfully posting the orders such that "close long position" and "close short position" trigger orders are correctly posted? I will really appreciate it. Thanks

**Comment by revilwang** [comment]
Sorry for the late reply. I'll provide a sample code:

```python
trigger = FuturesPriceTrigger(strategy_type=0, price_type=1, price=str(price), rule=1, expiration=0)
initial = FuturesInitialOrder(contract=symbol, size=10, price="0", tif="ioc")
futures_price_triggered_order = FuturesPriceTriggeredOrder(initial=initial, order_type="close-long-order", trigger=trigger)
```

The parameters are just randomly set. It only shows you how to initialize the order you needed.

**Comment by kasahh** [comment]
Hello, thanks for your response. I tried your sample with my parameters but none of the trigger take profit orders got posted, only my limit orders went through. I then tried inputing the following into your sample code respectively; 
1. changing size to 0
2. adding is_reduced_only = True
3. adding reduce_only = True
4. adding auto_size = close_long/auto_size = close_short

Each time the order got placed successfully unlike in your sample code but each time i end up with two "close long position" trigger orders like in my first screenshot.  I am beginning to wonder if the error is in the gateio python api itself. Maybe the "close short position" trigger order wasn't implemented? 

Perhaps, trying to recreate the error might help you understand better the issue i am facing. What I am trying to do is to place two limit orders for long and short and at the same time place two trigger take profit order for the long and short limit orders via gate io futures testnet. The long and short limit order always get posted but the trigger order always post two  "close long position" order rather than a "close long position" and a "close short position" trigger order. I have no idea why.
